### DISCUSSION TITLE
Testing fastapi infinite streams

### DISCUSSION THREAD

**Discussion by SH659** [encode/httpx#1889] [Potential Issue]
Is there any how-to?

I spend several hours reading source code and just have surprised. I stuck on [this line](https://github.com/encode/httpx/blob/master/httpx/_transports/asgi.py#L120). I just can't await my `client.send(req, stream=True)`  because it depends on the line I link above.

my stream is infinite e.g.
```
async def listen_raw():
    while True:
        yield b'1234\n'
        
        
@router.get('/raw')
async def recent_changes():
    return StreamingResponse(listen_raw())
    

@pytest.mark.asyncio
async def test_get_user_json_recent_changes(client: AsyncClient):
    req = client.build_request("GET", f"{settings.API_V1_STR}/recent_changes/raw")
    r = await client.send(req, stream=True)
    assert True # This code is unreachable
``` 

my env:
```
python==3.9
httpx==0.19.0
fastapi==0.70.0
```

**Comment by lovelydinosaur** [comment]
Ah, you'll be seeing this because the HTTPX ASGI transport *doesn't currently support streaming responses*.

(It always attempts to read the entire response, before returning.)

There's a PR that resolves this #998 - We we really ought to bring up to date with the latest version of `httpx` and get merged.
